src/EditorFeatures/CSharp/NamedArguments/NamedArgumentsAnalyzer.cs

```diff
@@ -0,0 +1,86 @@
+    {
+        public const string AddNamedArguments = nameof(AddNamedArguments);
+        public const string SortNamedArguments = nameof(SortNamedArguments);
+
+        private readonly static DiagnosticDescriptor s_addNamedArgumentsdescriptor = new DiagnosticDescriptor(
+            AddNamedArguments, CSharpEditorResources.AddNamedArguments, CSharpEditorResources.AddNamedArguments,
+            "Language", DiagnosticSeverity.Hidden, isEnabledByDefault: true);
+
+        private readonly static DiagnosticDescriptor s_sortNamedArgumentsdescriptor = new DiagnosticDescriptor(
+            SortNamedArguments, CSharpEditorResources.SortNamedArguments, CSharpEditorResources.SortNamedArguments,
+            "Language", DiagnosticSeverity.Hidden, isEnabledByDefault: true);
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(
+            s_addNamedArgumentsdescriptor, s_sortNamedArgumentsdescriptor);
+
+        public override void Initialize(AnalysisContext context)
+        {
+            // TODO: How about ObjectCreationExpression?
+            context.RegisterSyntaxNodeAction(AnalyseNode, SyntaxKind.InvocationExpression);
+        }
+
+        private static void AnalyseNode(SyntaxNodeAnalysisContext nodeContext)
+        {
+            var invocationExpr = (InvocationExpressionSyntax)nodeContext.Node;
+            var identifier = invocationExpr.Expression as IdentifierNameSyntax;
+
+            // TODO: When nameof is an actual method?
+            if (identifier?.Identifier.ValueText == "nameof")
+            {
+                return;
+            }
+
+            var arguments = invocationExpr.ArgumentList.Arguments;
+
+            // Any missing named argumets
+            if (arguments.Any(arg => arg.NameColon == null))
+            {
+                nodeContext.ReportDiagnostic(Diagnostic.Create(s_addNamedArgumentsdescriptor, invocationExpr.GetLocation()));
+            }
+
+            // Detect unordered named arguments
+            if (arguments.Any(arg => arg.NameColon != null))
+            {
+                var methodSymbol = nodeContext.SemanticModel.GetSymbolInfo(invocationExpr.Expression).Symbol as IMethodSymbol;
+                if (methodSymbol == null) return;
+
+                var e1 = arguments.GetEnumerator();
+                var e2 = methodSymbol.Parameters.GetEnumerator();
+
+                while (e1.MoveNext() && e2.MoveNext())
+                {
+                    var arg = e1.Current;
+                    var param = e2.Current;
+                    var nameColon = arg.NameColon?.Name.Identifier.ValueText;
+                    if (nameColon == null)
+                    {
+                        continue;
+                    }
+                    if (nameColon != param.Name)
+                    {
+                        nodeContext.ReportDiagnostic(Diagnostic.Create(s_sortNamedArgumentsdescriptor, invocationExpr.GetLocation()));
```

Comment:
I see.  it's not actually sorting.  It's matching the original declaration order.